File: C/24BCSH93/assignment9/question4.c
#include <stdio.h>

int checkPrime(int);

int main() {
	int num, flag;
	printf("Enter a number: ");
	scanf("%d", &num);
	flag = checkPrime(num);

	if (flag) {
		printf("It is a prime number\n");
	} else {
		printf("It is a composite number\n");
	}
	return 0;
}

int checkPrime(int num) {
	int i;

	for (i = 2; i <= num / 2; i++) {
		if (num % i == 0) {
			return 0;
		}
	}
	return 1;
}

File: MATLAB/assignment11/question4.m
%Q4. Input a number and check whether it is a prime or not.

n = input('Enter number: ');
flag = 1;

if n <= 1
    flag = 0;
else
    for i = 2:sqrt(n)
        if mod(n, i) == 0
            flag = 0;
            break;
        end
    end
end
if flag == 1
    disp('Prime number');
else
    disp('Not a prime number');
end
